vowpalwabbit/reductions/cb/cb_algs.cc

```diff
@@ -23,31 +23,43 @@ using namespace GEN_CS;
 
 namespace CB_ALGS
 {
-struct cb
+void generic_output_example(
+    VW::workspace& all, float loss, const VW::example& ec, const CB::label& ld, const CB::cb_class* known_cost)
 {
-  cb_to_cs cbcs;
-  VW::io::logger logger;
+  all.sd->update(ec.test_only, !CB::is_test_label(ld), loss, 1.f, ec.get_num_features());
 
-  cb(VW::io::logger logger) : logger(std::move(logger)) {}
-};
+  for (auto& sink : all.final_prediction_sink)
```

Comment:
nit:
```suggestion
  for (const auto& sink : all.final_prediction_sink)
```

?

Reply:
I always feel funny with a change like this because the const is not transitive to the contents of the unique ptr. Reading this look I'd probably make an assumption that whatever goes on inside the loop does not change final_prediction_sink which isn't really true